### ISSUE TITLE
Search Feature by Lunr

### ISSUE THREAD

**Issue by justinrummel** [mmistakes/hpstr-jekyll-theme#85]
Trying to integrate https://github.com/slashdotdash/jekyll-lunr-js-search to provide search results, I came into an interesting issue.  I believe the version of jQuery that is specified on the plugin conflicts with your jQuery which results in the Menu item not working.  My workaround was to create a specific /search/ page and if the search results don't supply with the correct items you wanted I display a "Go Home" button. 

Any thoughts how I can get the Menu item work sitewide, as I would like to have the capabilities to search on every page. 

Search page (with broken menu): https://www.justinrummel.com/search/ 

/search/index.md code
``` bash
---
layout: page
title: "Search"
date: 
modified:
description:
image:
  feature: abstract-3.jpg
  credit: dargadgetz
  creditlink: http://www.dargadgetz.com/ios-7-abstract-wallpaper-pack-for-iphone-5-and-ipod-touch-retina/
share: false
comments: false
---
<header class="post-header">
	<h1>Search: {{ site.title }}</h1>
</header>

<div id="search">
	<form action="/search" method="get">
		<input type="text" id="search-query" name="q" placeholder="Search" autocomplete="on">
	</form>
</div>

<section id="search-results" style="display: none;">
	<div align="right">(Can't find what you want?  Return <a href="{{ site.url }}">Home</a>)</div>
	<h1>Search results</h1>
	<div class="entries">
	</div>
</section>

{% raw %}
<script id="search-results-template" type="text/mustache">
	{{#entries}}
	<article>
		<h3>
			{{#date}}<small><time datetime="{{pubdate}}" pubdate>{{displaydate}}</time></small>{{/date}}
			- <a href="{{url}}">{{title}}</a>
		</h3>
	</article>
	{{/entries}}
</script>
{% endraw %}
```

scripts.html added code
``` bash
<!-- Search Results by jekyll-lunr-js-search -->
{% if page.title == "Search" %}
<script src="{{ site.url }}/assets/js/search.min.js" type="text/javascript" charset="utf-8"></script>
<script type="text/javascript">
$(function() {
	$('#search-query').lunrSearch({
		indexUrl: '/assets/js/index.json',           // URL of the `index.json` index data for your site
		results:  '#search-results',          // jQuery selector for the search results container
		entries:  '.entries',                 // jQuery selector for the element to contain the results list, must be a child of the results element above.
		template: '#search-results-template'  // jQuery selector for the Mustache.js template
	});
});
</script>
{% endif %}
```

**Comment by mmistakes** [comment]
I didn't write the dl-menu script so I can't speak to that or any sort of conflict it might be having with the search script.

Have you tried managing the Lunr dependencies yourself instead of using `js/search.min.js` that comes prebuilt from the gem? It looks like script bakes in the newer jQuery 2.1.1 which I'm guessing is the root of your conflict problem.

### As per Lunr, dependencies needed:

```
<script src="/js/jquery-1.9.1.min.js" type="text/javascript" charset="utf-8"></script>
<script src="/js/lunr.min.js" type="text/javascript" charset="utf-8"></script>
<script src="/js/mustache.js" type="text/javascript" charset="utf-8"></script>
<script src="/js/date.format.js" type="text/javascript" charset="utf-8"></script>
<script src="/js/URI.min.js" type="text/javascript" charset="utf-8"></script>
<script src="/js/jquery.lunr.search.js" type="text/javascript" charset="utf-8"></script>
```

As noted above jQuery 1.9.1 is a dependency for Lunr and happens to be same version I bundled into the theme (which does work with the menu). I'd try including those scripts manually instead of using `search.min.js` to see if that gives you a working menu and search.

Your other option would be to track down [the author the dl-menu script](https://github.com/codrops/ResponsiveMultiLevelMenu) and see if they've updated or have a fix for jQuery 2.1.1.

**Comment by justinrummel** [comment]
Thanks for the ideas!
